src/native/fs/fs_napi.cpp

```diff
@@ -646,6 +650,43 @@ struct FileWrite : public FSWorker
 };
 
 
+struct FileWritev : public FSWorker
+{
+    FileWrap *_wrap;
+    std::vector<struct iovec> iov_vec;
+    ssize_t _bw;
+    FileWritev(const Napi::CallbackInfo& info) 
+        : FSWorker(info)
+        , _bw(0)
+    {
+        _wrap = FileWrap::Unwrap(info.This().As<Napi::Object>());
+        auto buffers = info[1].As<Napi::Array>();
+        const int buffers_len = buffers.Length();
```

Comment:
Since the length is known, I would resize the vector to the length here, and then directly update the elements instead of push_back one by one potentially causing dynamic allocations and reallocations...